docs/runbooks/VirtualMachineCRCErrors.md

```diff
@@ -14,26 +14,54 @@ major service outages.
 
 ## Diagnosis
 
-1) Get the volume name from the VM:
+Execute the following bash script to get a list of Virtual Machines that are
+using a storage class that is incorrectly configured:
 
 ```bash
-$ kubectl get vm <vm-name> -o jsonpath='{.spec.template.spec.volumes}'
+#!/bin/bash
+
+# if NAMESPACE is set, Get all PVCs in the namespace that are owned by a VM
+if [[ -n "$NAMESPACE" ]]; then
+  pvcs=$(kubectl get pvc -n "$NAMESPACE" -o=jsonpath='{range .items[?(@.metadata.ownerReferences[0].kind=="VirtualMachine")]}{.metadata.namespace}/{.metadata.name}{"\n"}{end}')
+else
+  # Get all PVCs in all namespaces that are owned by a VM
+  pvcs=$(kubectl get pvc --all-namespaces -o=jsonpath='{range .items[?(@.metadata.ownerReferences[0].kind=="VirtualMachine")]}{.metadata.namespace}/{.metadata.name}{"\n"}{end}')
+fi
+
+results=""
+
+# Loop through each PVC to check for the issue
+count=0
+issues=0
+total=$(echo "$pvcs" | wc -l)
+while IFS= read -r pvc_info; do
+  count=$((count+1))
+  echo "Checking PVC $pvc_info ($count/$total)"
+    
+  namespace=$(echo "$pvc_info" | cut -d'/' -f1)
+  pvc_name=$(echo "$pvc_info" | cut -d'/' -f2)
+  
+  # Get VM name and PV name
+  read -r vm_name pv_name <<< $(kubectl get pvc "$pvc_name" -n "$namespace" -o=jsonpath='{.metadata.ownerReferences[0].name} {.spec.volumeName}')
+  
+  # Get storage class, csi driver, volume mode, and map options
+  read -r storage_class csi_driver volume_mode map_options <<< $(kubectl get pv "$pv_name" -o=jsonpath='{.spec.storageClassName} {.spec.csi.driver} {.spec.volumeMode} {.spec.csi.volumeAttributes.mapOptions}')
+  
+  # Check conditions
+  if [[ "$volume_mode" == "Block" && "$csi_driver" == *"rbd.csi.ceph.com"* && "$map_options" != *"krbd:rxbounce"* ]]; then
+      results="$results\n$namespace | $vm_name | $pvc_name | $storage_class"
+      issues=$((issues+1))
+  fi
+done <<< "$pvcs"
+
+if [[ -z "$results" ]]; then
+  echo "No Virtual Machines found with the issue"
+else 
+  echo -e "\n\n$issues issues found:\nNamespace | Virtual Machine | PVC | Storage Class"
+  echo -e "$results"
+fi
 ```
 
-2) Get the storage class name from the volume:
-
-```bash
-$ kubectl get pvc <volume-name> -o jsonpath='{.spec.storageClassName}'
-```
-
-3) Get the storage class configuration:
-
-```bash
-$ kubectl get sc <storage-class-name> -o yaml
-```
-
-4) Check if the storage class has the "krbd:rxbounce" map option:
-
```

Comment:
An elipsis in a YAML file should be `# ...`, not `...`.

Reply:
updated